### PR TITLE
[TableGen] [NFC] Remove unused includes in TableGen BE

### PR DESCRIPTION
**Pull request by jerryyiransun**
split PR as requested from https://github.com/llvm/llvm-project/pull/113318.

Removes unused imports in TableGen BE

### CODE REVIEW

File: llvm/utils/TableGen/WebAssemblyDisassemblerEmitter.cpp [hunk 1]
```diff
@@ -15,7 +15,6 @@
 
 #include "WebAssemblyDisassemblerEmitter.h"
 #include "Common/CodeGenInstruction.h"
-#include "llvm/ADT/STLExtras.h"
```
Comment (kparzysz) [inline llvm/utils/TableGen/WebAssemblyDisassemblerEmitter.cpp:18]: There is a use of `llvm::append_range` on line 136.

File: llvm/utils/TableGen/InstrInfoEmitter.cpp [hunk 2]
```diff
@@ -24,7 +24,6 @@
 #include "llvm/ADT/STLExtras.h"
 #include "llvm/ADT/SmallVector.h"
 #include "llvm/ADT/StringExtras.h"
-#include "llvm/Support/Casting.h"
```
Comment (kparzysz) [inline llvm/utils/TableGen/InstrInfoEmitter.cpp:27]: This file does use `cast` etc.  The header is rarely included directly, but if we follow the IWYU principle, it should probably be included more often.

File: llvm/utils/TableGen/CTagsEmitter.cpp [hunk 3]
```diff
@@ -17,7 +17,6 @@
 #include "llvm/TableGen/Error.h"
 #include "llvm/TableGen/Record.h"
 #include "llvm/TableGen/TableGenBackend.h"
-#include <algorithm>
 #include <vector>
```
Comment (jurahul) [inline llvm/utils/TableGen/CTagsEmitter.cpp:20]: Can we add <tuple> as it seems the file uses std::tuple?

**Comment by jurahul** [comment]
Minor comment: import is not a C++ term. Can you change the description to "Remove unused includes in TableGen BE" (note change in capitalization as well).

**Comment by arsenm** [comment]
> Minor comment: import is not a C++ term. 

It is actually, but we're not using modules yet

**Review by jurahul — APPROVED** [review]

**Pull request merged**